allure-testng/src/test/java/io/qameta/allure/test/FeatureCombinationsTest.java

```diff
@@ -0,0 +1,326 @@
+        final ClassLoader classLoader = getClass().getClassLoader();
+        List<String> suiteFiles = Arrays.stream(suites)
+                .map(classLoader::getResource)
+                .map(URL::getFile)
+                .collect(Collectors.toList());
+        assertThat(suites).as("Cannot find all suite xml files").hasSameSizeAs(suiteFiles);
+        testNG.setTestSuites(suiteFiles);
+        testNG.run();
+    }
+
+    @Test
+    public void singleTest() throws IOException {
+        final String testName = "testWithOneStep";
+        runTestNGSuites("suites/single-test.xml");
+        List<TestResult> testResult = results.getTestResults();
+
+        assertThat(testResult).as("Test case result has not been written").hasSize(1);
+        assertThat(testResult.get(0)).as("Unexpected passed test property")
+                .hasFieldOrPropertyWithValue("status", Status.PASSED)
+                .hasFieldOrPropertyWithValue("stage", Stage.FINISHED)
+                .hasFieldOrPropertyWithValue("name", testName);
+        assertThat(testResult)
+                .flatExtracting(TestResult::getSteps)
+                .hasSize(1)
+                .flatExtracting(StepResult::getStatus)
+                .contains(Status.PASSED);
+    }
+
+    @Test
+    public void failingByAssertion() {
+        String testName = "failingByAssertion";
+        runTestNGSuites("suites/failing-by-assertion.xml");
+        List<TestResult> testResult = results.getTestResults();
+
+        assertThat(testResult).as("Test case result has not been written").hasSize(1);
+        assertThat(testResult.get(0)).as("Unexpected failed test property")
+                .hasFieldOrPropertyWithValue("status", Status.FAILED)
+                .hasFieldOrPropertyWithValue("stage", Stage.FINISHED)
+                .hasFieldOrPropertyWithValue("name", testName);
+        assertThat(testResult)
+                .flatExtracting(TestResult::getSteps)
+                .hasSize(2)
+                .flatExtracting(StepResult::getStatus)
+                .contains(Status.PASSED, Status.FAILED);
+    }
+
+    @Test
+    public void brokenTest() {
+        String testName = "brokenTest";
+        runTestNGSuites("suites/broken.xml");
+        List<TestResult> testResult = results.getTestResults();
+
+        assertThat(testResult).as("Test case result has not been written").hasSize(1);
+        assertThat(testResult.get(0)).as("Unexpected broken test property")
+                .hasFieldOrPropertyWithValue("status", Status.BROKEN)
+                .hasFieldOrPropertyWithValue("stage", Stage.FINISHED)
+                .hasFieldOrPropertyWithValue("name", testName);
+        assertThat(testResult)
+                .flatExtracting(TestResult::getSteps)
+                .hasSize(2)
+                .flatExtracting(StepResult::getStatus)
+                .contains(Status.PASSED, Status.BROKEN);
+    }
+
+    @Test
+    public void beforeFixturesCombination() {
```

Comment:
think it should be splitted to separate cases  - it will reduce the number of unrelated vars and helps to understand cases. Maybe try to use nested static classes to group them